src/target/source/codegen_c.cc

```diff
@@ -631,8 +632,11 @@ void CodeGenC::PrintVecBinaryOp(const std::string& op, DataType t, PrimExpr lhs,
   }
 }
 
+unsigned int random_seed = 0;
 void CodeGenC::VisitStmt_(const AllocateConstNode* op) {
-  std::string symbol_name = op->buffer_var->name_hint;
+  // Add a random suffix to eliminate duplicate global variables.
+  int suffix = rand_r(&random_seed) % (2 << 24);
```

Comment:
any reason to not just use a counter and add that as suffix to each global var name?

Reply:
A counter would work fine - this is just a hack to fix the duplicate global variables problem. Would love @areusch's take on the right way to fix this long term.

Reply:
I believe a similar fix is needed for the case usmp is enabled here: https://github.com/apache/tvm/blob/2877c5a3cf126637f0968bb9090454410c426cd0/src/target/source/source_module.cc#L295